client/my-sites/design-menu/design-tools/header-image-control.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import get from 'lodash/get';
+import assign from 'lodash/assign';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import ImagePreloader from 'components/image-preloader';
+import Button from 'components/button';
+import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
+import EditorMediaModal from 'post-editor/media-modal';
+
+const HeaderImageControl = React.createClass( {
```

Comment:
That was my original intention, actually. I had originally only had the
props for their settings and an onChange function, but then I added the
mapStateToProps here to simplify the container. I may move that back into
the container so this can remain pure.
On Mon, Mar 21, 2016 at 9:00 AM Matias Ventura <notifications@github.com>
wrote:

> In client/my-sites/design-menu/design-tools/header-image-control.js
> <https://github.com/Automattic/wp-calypso/pull/4156#discussion_r56817488>:
>
> > +import React from 'react';
> > +import { connect } from 'react-redux';
> > +import get from 'lodash/get';
> > +import assign from 'lodash/assign';
> > +
> > +/**
> > + * Internal dependencies
> > + */
> > +import FormFieldset from 'components/forms/form-fieldset';
> > +import FormLabel from 'components/forms/form-label';
> > +import ImagePreloader from 'components/image-preloader';
> > +import Button from 'components/button';
> > +import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
> > +import EditorMediaModal from 'post-editor/media-modal';
> > +
> > +const HeaderImageControl = React.createClass( {
>
> For instance, this would be a header-image component outside of design
> menu, just a pure calypso component we can drop and use anywhere, since
> preview is decoupled.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Automattic/wp-calypso/pull/4156/files/b0478d0b8124758a2f3b9e770577088614aa756c#r56817488>
>